src/components/dashboard/dashboardItem/DataConsumption.js

```diff
@@ -0,0 +1,184 @@
+/**
+ *
+ * @author sriram
+ *
+ * A chart that displays the data consumption of the user.
+ *
+ */
+
+import React from "react";
+import { useTranslation } from "i18n";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import { useQuery } from "react-query";
+import {
+    Chart as ChartJS,
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend,
+} from "chart.js";
+import { Bar } from "react-chartjs-2";
+import { getDataUsage, DATA_USAGE_QUERY_KEY } from "serviceFacades/dashboard";
+import constants from "../../../constants";
+import { formatFileSize } from "components/data/utils";
+import ErrorTypographyWithDialog from "components/error/ErrorTypographyWithDialog";
+import { getErrorCode } from "components/error/errorCode";
+import { Typography, useTheme } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+import { getUnixTime, parseISO } from "date-fns";
+import {
+    getFormattedDistance,
+    formatDateObject,
+} from "components/utils/DateFormatter";
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend,
+    ChartDataLabels
+);
+
+const options = (usage, quota, date, distance, title, theme, t) => {
+    return {
+        indexAxis: "y",
+        plugins: {
+            title: {
+                display: true,
+                text: [title, date],
+            },
+            legend: {
+                display: false,
+            },
+            tooltip: {
+                callbacks: {
+                    label: function (context) {
+                        let label = context.dataset.label || "";
+                        return label;
+                    },
+                    footer: function (context) {
+                        return t("dataConsumptionTimestamp", { distance });
+                    },
+                },
+                xAlign: "right",
+                yAlign: "top",
+            },
+            datalabels: {
+                display: () => (100 * usage) / quota > 20, //if usage > 10% of quota, show data labels
+                backgroundColor: function (context) {
+                    return context.dataset.backgroundColor;
+                },
+                borderRadius: 4,
+                color: theme.palette.primary.contrastText,
+                font: {
+                    weight: "bold",
+                },
+                formatter: (value) =>
+                    `${formatFileSize(value)} of ${formatFileSize(quota)} `,
+                padding: 1,
+            },
+        },
+        responsive: false,
+        scales: {
+            x: {
+                stacked: false,
+                min: 0,
+                max: usage < quota ? quota : usage,
+                ticks: {
+                    /*  stepSize:
+                         usage < quota ? 10 : Math.ceil(usage / quota) * 10, */
```

Comment:
I know this is commented out for now (or should it just be removed now?), but for future reference, `Math.max` might work here as well:
```suggestion
                    /*  stepSize: Math.max(10, Math.ceil(usage / quota) * 10), */
```